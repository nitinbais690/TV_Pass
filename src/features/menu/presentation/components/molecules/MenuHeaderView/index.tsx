import { useLocalization } from 'contexts/LocalizationContext';
import React from 'react';
import { View } from 'react-native';
import PageTitle from '../../atoms/PageTitle';
import MenuHeaderViewStyle from './style';
import CloseIcon from 'assets/images/close_icon.svg';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { AUTOMATION_TEST_ID } from 'features/menu/presentation/automation-ids';

/*
   Renders Menu Header view - Page title with close button
 */
export default function MenuHeaderView(props: MenuHeaderProps) {
    const { strings } = useLocalization();

    const styles = MenuHeaderViewStyle();

    return (
        <View style={styles.container}>
            <PageTitle title={props.showTitle ? strings['tabs.menu'].toUpperCase() : ''} />
            <TouchableOpacity
                onPress={props.closeBtnAction}
                testID={AUTOMATION_TEST_ID.CLOSE_ICON}
                accessibilityLabel={AUTOMATION_TEST_ID.CLOSE_ICON}>
                <CloseIcon
                    width={styles.closeBtnStyle.width}
                    height={styles.closeBtnStyle.height}
                    style={styles.closeBtnStyle}
                />
            </TouchableOpacity>
        </View>
    );
}

interface MenuHeaderProps {
    showTitle: boolean;
    closeBtnAction: () => void;
}
