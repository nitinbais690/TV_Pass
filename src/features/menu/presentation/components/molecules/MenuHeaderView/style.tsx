import { scale } from 'qp-common-ui';
import { appPaddingValues } from 'core/styles/AppStyles';
import { StyleSheet } from 'react-native';

export default function MenuHeaderViewStyle() {
    return StyleSheet.create({
        container: {
            flexDirection: 'row',
        },
        closeBtnStyle: {
            marginTop: scale(47),
            marginStart: appPaddingValues.lg,
            marginEnd: appPaddingValues.lg,
            width: scale(24),
            height: scale(24),
        },
    });
}
