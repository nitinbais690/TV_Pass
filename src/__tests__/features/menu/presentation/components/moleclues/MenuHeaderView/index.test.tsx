import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import MenuHeaderView from 'features/menu/presentation/components/molecules/MenuHeaderView';

jest.mock('core/presentation/hooks/use-app-colors');

describe('Menu header view test', () => {
    it('Menu header view render correctly', () => {
        const pressHandler = jest.fn();
        const { container } = render(<MenuHeaderView closeBtnAction={pressHandler} />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
