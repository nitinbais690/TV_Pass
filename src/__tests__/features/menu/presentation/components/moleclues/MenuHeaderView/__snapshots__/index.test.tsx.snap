// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu header view test Menu header view render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginEnd": 28,
          "marginStart": 28,
          "marginTop": 68.48571428571428,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "ProximaNova-Black",
              "fontSize": 44,
              "fontWeight": "900",
            },
            Object {
              "color": "#ECECEC",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <RNGestureHandlerButton
      collapsable={false}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityLabel="CLOSE_ICON"
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="CLOSE_ICON"
      >
        <SvgMock
          height={34.971428571428575}
          style={
            Object {
              "height": 34.971428571428575,
              "marginEnd": 28,
              "marginStart": 28,
              "marginTop": 68.48571428571428,
              "width": 34.971428571428575,
            }
          }
          width={34.971428571428575}
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
